client/ayon_houdini/api/workfile_template_builder.py

```diff
@@ -0,0 +1,287 @@
+
+        fill_data["root"] = anatomy.roots
+        fill_data["project"] = {
+            "name": project_name,
+            "code": anatomy.project_code,
+        }
+
+        result = StringTemplate.format_template(path, fill_data)
+        if result.solved:
+            path = result.normalized()
+
+        # I copied the whole thing because I wanted to add some
+        # Houdini specific code here
+        path = hou.text.expandString(path)
+
+        if path and os.path.exists(path):
+            self.log.info("Found template at: '{}'".format(path))
+            return {
+                "path": path,
+                "keep_placeholder": keep_placeholder,
+                "create_first_version": create_first_version
+            }
+
+        solved_path = None
+        while True:
+            try:
+                solved_path = anatomy.path_remapper(path)
+            except KeyError as missing_key:
+                raise KeyError(
+                    "Could not solve key '{}' in template path '{}'".format(
+                        missing_key, path))
+
+            if solved_path is None:
+                solved_path = path
+            if solved_path == path:
+                break
+            path = solved_path
+
+        solved_path = os.path.normpath(solved_path)
+        if not os.path.exists(solved_path):
+            raise TemplateNotFound(
+                "Template found in AYON settings for task '{}' with host "
+                "'{}' does not exists. (Not found : {})".format(
+                    task_name, host_name, solved_path))
+
+        self.log.info("Found template at: '{}'".format(solved_path))
+
+        return {
+            "path": solved_path,
+            "keep_placeholder": keep_placeholder,
+            "create_first_version": create_first_version
+        }
+
+    def import_template(self, path):
+        """Import template into current scene.
+        Block if a template is already loaded.
+
+        Args:
+            path (str): A path to current template (usually given by
+            get_template_preset implementation)
+
+        Returns:
+            bool: Whether the template was successfully imported or not
+        """
+
+        # TODO Check if template is already imported
+
+        # Load template workfile
+        self.host.open_workfile(path)
+
+        return True
+
+
+class HoudiniPlaceholderPlugin(PlaceholderPlugin):
+    """Base Placeholder Plugin for Houdini with one unified cache.
+
+    Inherited classes must still implement `populate_placeholder`
+    """
+
+    def get_placeholder_node_name(self, placeholder_data):
+        return self.identifier.replace(".", "_")
+    
+    def create_placeholder_node(self, node_name=None):
+        """Create node to be used as placeholder.
+
+        By default, it creates a null node in '/out'.
+        Feel free to override it in different workfile build plugins.
+        """
+ 
+        node = hou.node("/out").createNode("null", node_name)
```

Comment:
You might want to add the `moveToGoodPosition` call on the created node because it seems to stack all newly created nodes right on top of each other.

Reply:
added it here https://github.com/ynput/ayon-houdini/pull/36/commits/c92791c10661a80fb06ba3c6bb2ee48506f6d054